### PR TITLE
Cleanup GraalVM metadata and used Metadata Repository

### PR DESCRIPTION
**Pull request by graemerocher**
With the GraalVM Metadata Repository now active we no longer need metadata for Netty and Logback so this PR removes that metadata. In addition this PR removes the use of all GraalVM internal APIs from core and upgrades to the latest version of SVM.

Finally, this PR deprecates `AutomaticFeatureUtils` because in future it should not be necessary to use this class with metadata living the repository.

### CODE REVIEW

**Review by melix — APPROVED** [review]
Fingers crossed that it passes tests!

**Comment by graemerocher** [comment]
Tests passed 🕺

**Comment by graemerocher** [comment]
Hold off merging, the logback metadata is incomplete in the repository

**Comment by graemerocher** [comment]
ok fixed logging. For Launch we have to remove the line:

```xml
<withJansi>true</withJansi>
```

Since JANSI is not the classpath anyway

File: .github/workflows/graalvm.yml [hunk 1]
```diff
@@ -19,14 +19,8 @@ jobs:
...
     strategy:
       matrix:
-        graalvm: [ 'latest', 'dev' ]
+        graalvm: [ 'latest']
         java: [ '17' ]
-        include:
-          - graalvm: 'dev'
-            java: '19'
-        exclude:
-          - graalvm: 'dev'
```
Comment (fniephaus) [inline .github/workflows/graalvm.yml:28]: JDK17-based dev builds are back online 🙂

**Review by sdelamo — APPROVED** [review]
see: https://github.com/micronaut-projects/micronaut-core/pull/8824

**Pull request merged**